model/action.go

```diff
@@ -41,6 +41,21 @@ type Action struct {
 	Condition string `json:"condition,omitempty"`
 }
 
+type actionForUnmarshal Action
+
+// UnmarshalJSON ...
+func (a *Action) UnmarshalJSON(data []byte) error {
```

Comment:
for default values you can also register a struct validation to set it, e.g.:

```
func init() {
	val.GetValidator().RegisterStructValidation(ActionStructLevelValidation, Action{})
}

func EventStructLevelValidation(structLevel validator.StructLevel) {
	eactionent := structLevel.Current().Interface().(Action)

if len(action) == 0.... do something..
}
```

Reply:
I would think adding a validator this way would make the separation between the unmarshalling process and the setting of default value even farther apart making the logic scattered all around the repository(debugging and maintaining harder). The current way implementation looks good as the first place people look at would be the struct and they would be able to see the default value "close-by"

Reply:
I agree, @brlala. We can keep the default setting in the unmarshalling process. We already do this in other SDKs, like the Java one IIRC.